h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/rabit/RabitTrackerH2O.java

```diff
@@ -93,13 +98,26 @@ public void stop() {
 
         private LinkMap linkMap;
         private Map<String, Integer> jobToRankMap = new HashMap<>();
+        private List<SocketChannel> socketChannels = new ArrayList<>();
 
         private RabitTrackerH2OThread(RabitTrackerH2O tracker) {
             setPriority(MAX_PRIORITY-1);
             this.setName("TCP-" + tracker.sock);
             this.tracker = tracker;
         }
 
+        @Override
+        public void interrupt() {
+            super.interrupt();
```

Comment:
We need to make sure this is thread-safe. interrupt() can be called by the thread itself and also by any other thread. The Tracker thread can be modifying the list while another thread calls interrupt().

Do we need to address this? WDYT?